### DISCUSSION TITLE
API Requests in Middleware

### DISCUSSION THREAD

**Discussion by biapessoab** [vercel/next.js#65065] [Help]
### Summary

hey, im currently struggling with middleware usage. i need to set the headers authorization of my api requests and i want to do it using middleware, but the request dont even are identified by middleware. actually, some of them are, most _next/. but specifically my api requests doesnt get recognized. here is my code:

```
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";

export function middleware(request: NextRequest) {
  const res = NextResponse.next();
  const newHeaders = new Headers(request.headers)

  newHeaders.set('Authorization', 'header')

  console.log("Path: " + request.nextUrl.pathname)
  console.log()

  return NextResponse.next({
    request: {
      headers: newHeaders,
    }
  })
};
```

**Comment by icyJoseph** [comment]
Are they API request against another domain? Only requests to your own server are intercepted by middleware.
